chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -397,199 +399,207 @@
-      case
-      when os =~ /(ubuntu|debian)/
-        bin="deb http://#{admin_ip}:#{web_port}/#{os}/crowbar-extra/#{f} /"
-        src="deb-src http://#{admin_ip}:#{web_port}/#{os}/crowbar-extra/#{f} /"
-        node.set[:provisioner][:repositories][os][f][bin] = true if
-          File.exists? "#{os_dir}/crowbar-extra/#{f}/Packages.gz"
-        node.set[:provisioner][:repositories][os][f][src] = true if
-          File.exists? "#{os_dir}/crowbar-extra/#{f}/Sources.gz"
-      when os =~ /(redhat|centos|suse)/
-        bin="baseurl=http://#{admin_ip}:#{web_port}/#{os}/crowbar-extra/#{f}"
-        node.set[:provisioner][:repositories][os][f][bin] = true
-        else
-          raise ::RangeError.new("Cannot handle repos for #{os}")
-        end
+node[:provisioner][:supported_oses].each do |os, arches|
+  arches.each do |arch, params|
+    web_path = "#{provisioner_web}/#{os}/#{arch}"
+    admin_web = "#{web_path}/install"
+    crowbar_repo_web = "#{web_path}/crowbar-extra"
+    os_dir = "#{tftproot}/#{os}/#{arch}"
+    os_codename = node[:lsb][:codename]
+    role = "#{os}_install"
+    missing_files = false
+    append = params["append"].dup # We'll modify it inline
+    initrd = params["initrd"]
+    kernel = params["kernel"]
+    require_install_dir = params["require_install_dir"].nil? ? true : params["require_install_dir"]
+
+    if require_install_dir
+      # Don't bother for OSes that are not actually present on the provisioner node.
+      next unless File.directory? os_dir and File.directory? "#{os_dir}/install"
     end
-  end
 
-  # If we were asked to use a serial console, arrange for it.
-  if node[:provisioner][:use_serial_console]
-    append << " console=tty0 console=#{node[:provisioner][:serial_tty]}"
-  end
-
-  # Make sure we get a crowbar install key as well.
-  unless crowbar_key.empty?
-    append << " crowbar.install.key=#{crowbar_key}"
-  end
-
-  # These should really be made libraries or something.
-  case
-  when /^(suse)/ =~ os
-    # Add base OS install repo for suse
-    node.set[:provisioner][:repositories][os]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install" => true }
-
-    ntp_servers = search(:node, "roles:ntp-server")
-    ntp_servers_ips = ntp_servers.map { |n| Chef::Recipe::Barclamp::Inventory.get_network_by_type(n, "admin").address }
-
-    target_platform_version = os.gsub(/^.*-/, "")
-    template "#{os_dir}/crowbar_join.sh" do
-      mode 0644
-      owner "root"
-      group "root"
-      source "crowbar_join.suse.sh.erb"
-      variables(admin_ip: admin_ip,
-                web_port: web_port,
-                ntp_servers_ips: ntp_servers_ips,
-                target_platform_version: target_platform_version)
+    # Index known barclamp repositories for this OS
+    node[:provisioner][:repositories][os] ||= Mash.new
+    node[:provisioner][:repositories][os][arch] ||= Mash.new
+
+    if File.exists? "#{os_dir}/crowbar-extra" and File.directory? "#{os_dir}/crowbar-extra"
```

Comment:
Style/AndOr: Use `&&` instead of `and`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)